### PR TITLE
Rerun Kubelet Scraper When Data is Populated

### PR DESCRIPTION
**Pull request by xqi-nr**
This PR addresses the issue when Kubelet becomes temporarily unavailable, no Kubelet metrics are populated, and the error "kubelet data was not populated after trying all endpoints" from Kubelet scraper is returned to main.  
The current behavior of K8s agent main is that if any (ksm, kubelet, controlplane) scrapers returns any error, the K8s agent will Immediately fail and exit with code 5.

Instead of exiting immediately when kubelet is not reachable, we can set a ScraperMaxReruns (e.g., 4 times) at the scraper level. This will improve customer experience when kubelet is temporarily not reachable.

### CODE REVIEW

File: cmd/nri-kubernetes/main.go
Comment (nserrino) [inline cmd/nri-kubernetes/main.go:198]: looks like we can omit the else statement since the if above has a return? can help readability a bit. overall looks good!

Comment (xqi-nr) [reply cmd/nri-kubernetes/main.go]: thanks, remove the else

**Review by nserrino — APPROVED** [review]

File: charts/newrelic-infrastructure/Chart.yaml
Comment (htroisi) [inline charts/newrelic-infrastructure/Chart.yaml:12]: If we make changes to the agent, I believe we want to bump both the `appVersion` and the chart `version`. 

That said, I am new to helm charts and the `nri-kubernetes` release process, so I'm less clear on how this works in practice. 😅  Do we want to make these chart changes in this PR or wait until we do a release next Wednesday?

Comment (xqi-nr) [reply charts/newrelic-infrastructure/Chart.yaml:12]: @nserrino I made several updates to address the lints, including the version change, after you approval. :) So pls review the PR again. I will revert the version change if it is not proper.

Comment (xqi-nr) [reply charts/newrelic-infrastructure/Chart.yaml]: I reverted the chart change before the PR merge

**Pull request merged**
